File: python/sort/selection.py
def sort(data: list) -> None:
    for i in range(len(data)):
        lowest_index = i
        for j in range(i + 1, len(data)):
            if data[lowest_index] > data[j]:
                lowest_index = j
        data[i], data[lowest_index] = data[lowest_index], data[i]

File: rust/src/sort/selection.rs
/// Sort the data using selection sort
pub fn sort<T: PartialOrd>(data: &mut [T]) {
    // Iterate through the list `data.len()` times
    for i in 0..data.len() {
        // Use the current iteration count as initial minimum index
        let mut min_index = i;

        // Find the actual lowest index
        for j in i + 1..data.len() {
            if data[min_index] > data[j] {
                min_index = j;
            }
        }

        // Swap the lowest element with the element at the current index
        data.swap(i, min_index);
    }
}
